Libraries/Azure.psm1

```diff
@@ -213,57 +205,179 @@ Function Validate-SubscriptionUsage($RGXMLData, $Location, $OverrideVMSize, $Sto
-        } elseif ($storageAccount -imatch "NewStorage_Standard") {
-            $changedSC = "NewStorage_Premium_LRS"
-        } elseif (($storageAccount -imatch "NewStorage_Premium") -or ($storageAccount -imatch "ExistingStorage_Premium")) {
-            $changedSC = ""
-        } else {
-            $current_sc = Get-StorageAccountFromRegion -Region $Location -StorageAccount "ExistingStorage_Standard"
-            if ($current_sc -eq $storageAccount) {
-                $changedSC = Get-StorageAccountFromRegion -Region $Location -StorageAccount "ExistingStorage_Premium"
-                if ($OsVHD) {
-                    $copyVHD = $true
-                }
-            } else {
-                $storageAccountType = (Get-AzStorageAccount | Where-Object {$_.StorageAccountName -eq $storageAccount}).Sku.Tier.ToString()
-                if ($storageAccountType -inotmatch "premium") {
-                    Write-LogErr "Provided storage account is not premium type, this case $($TestCaseData.testName) need run under premium type of storage account."
-                    Throw "Case $($TestCaseData.testName) need run under premium type of storage account."
-                }
-            }
-        }
-    }
+	# From PrepareTestEnvironment(), $GlobalConfig.Global.Azure.Subscription.ARMStorageAccount has already been existing Storage Account Name
+	$storageAccount = $GlobalConfig.Global.Azure.Subscription.ARMStorageAccount
+	$changedSC = $storageAccount
+	if ($TestCaseData.AdditionalHWConfig.StorageAccountType -and $TestCaseData.AdditionalHWConfig.StorageAccountType.Contains("Premium")) {
+		$storageAccountType = (Get-AzStorageAccount | Where-Object {$_.StorageAccountName -eq $storageAccount}).Sku.Tier.ToString()
+		if ($storageAccountType -inotmatch "premium") {
+			$current_sc = Get-StorageAccountFromRegion -Region $Location -StorageAccount "ExistingStorage_Standard"
+			# if $storageAccount from GlobalConfig is the existing standard storage accounts, then change to existing Premium storage
+			if ($current_sc -eq $storageAccount) {
+				$changedSC = Get-StorageAccountFromRegion -Region $Location -StorageAccount "ExistingStorage_Premium"
+				if ($OsVHD) {
+					$sourceContainer =  $osVHD.Split("/")[$osVHD.Split("/").Count - 2]
+					$vhdName = $osVHD.Split("?")[0].split('/')[-1]
+					Write-LogInfo "Copy VHD from $storageAccount to $changedSC."
+					if(($OsVHD -imatch 'sp=') -and ($OsVHD -imatch 'sig=')) {
+						$copyStatus = Copy-VHDToAnotherStorageAccount -SasUrl $osVHD -destinationStorageAccount $changedSC `
+							-destinationStorageContainer "vhds" -vhdName $vhdName
+					} else {
+						$copyStatus = Copy-VHDToAnotherStorageAccount -sourceStorageAccount $storageAccount -sourceStorageContainer $sourceContainer `
+							-destinationStorageAccount $changedSC -destinationStorageContainer "vhds" -vhdName $vhdName
+					}
+					if (!$copyStatus) {
+						Throw "Failed to copy the VHD $storageAccount to $changedSC."
```

Comment:
Have you exercised 252 and 259 throw statements? I am not sure its exception ends to the run-time.

Reply:
No, I haven't. Just keeping previous code logic and coding implementation. You can check the change details.